internal/backends/mcu/pico_st7789.rs

```diff
@@ -153,16 +185,89 @@ pub fn init() {
         vec![Rgb565::default(); DISPLAY_SIZE.width as _].leak(),
         stolen_spi,
     );
-
-    crate::init_with_display(PicoDevices {
+    let mut buffer_provider = DrawBuffer {
         display,
-        touch,
-        last_touch: Default::default(),
-        timer,
         buffer: vec![Rgb565::default(); DISPLAY_SIZE.width as _].leak(),
         pio: Some(pio),
         stolen_pin: (dc_copy, cs_copy),
-    });
+        dirty_region: Default::default(),
+    };
+
+    let mut last_touch = None;
+    loop {
+        i_slint_core::timers::update_timers();
+        if NEEDS_REDRAW.load(Ordering::Relaxed) {
+            NEEDS_REDRAW.store(false, Ordering::Relaxed);
+            if let Some(window) = WINDOW.with(|x| x.borrow().clone()) {
+                let runtime_window = window.self_weak.upgrade().unwrap();
+                runtime_window.update_window_properties();
+
+                let scale_factor = runtime_window.scale_factor();
+                let size = DISPLAY_SIZE.to_f32() / scale_factor;
+                runtime_window.set_window_item_geometry(size.width as _, size.height as _);
```

Comment:
Hm, this is done on every draw now. I guess it doesn't hurt because `Property::set` won't do anything if the value hasn't changed.

Reply:
Right.

But we still need to do something about the size and other properties